apis/Google.Devtools.AspNet/Google.Devtools.AspNet/CloudErrorReporting/CloudErrorReportingExceptionLogger.cs

```diff
@@ -94,15 +100,31 @@ public sealed class CloudErrorReportingExceptionLogger : ExceptionLogger
         /// <inheritdoc />
         public override Task LogAsync(ExceptionLoggerContext context, CancellationToken cancellationToken)
         {
+            if (IncompleteClientAndTooManyPendingTasks())
+            {
+                return Task.FromResult(false);
+            }
+
             ReportedErrorEvent errorEvent = CreateReportRequest(context);
-            return _client.ReportErrorEventAsync(_projectResourceName, errorEvent);
+            return _clientTask.ContinueWith(clientTask =>
+            {
+                return clientTask.Result.ReportErrorEventAsync(_projectResourceName, errorEvent);
+            });
         }
 
         /// <inheritdoc />
         public override void Log(ExceptionLoggerContext context)
         {
+            if (IncompleteClientAndTooManyPendingTasks())
+            {
+                return;
+            }
+
             ReportedErrorEvent errorEvent = CreateReportRequest(context);
-            _client.ReportErrorEvent(_projectResourceName, errorEvent);
+            _clientTask.ContinueWith(clientTask => 
```

Comment:
I think we should throw.  If the client has faulted we have a problem.

I just updated this file to minimize churn.